tests/functional_tests/p2p.py

```diff
@@ -175,12 +174,25 @@ def test_p2p_tx_propagation(self):
         assert len(res.tx_hash) == 32*2
         txid = res.tx_hash
 
-        time.sleep(5)
-
-        for daemon in [daemon2, daemon3]:
+        # Due to Dandelion++, the network propagates transactions with a
+        # random delay, so poll for the transaction with a timeout
+        timeout = 16
```

Comment:
@vtnerd Do you know the expected maximum propagation delay in this scenario, based on the D++ paper? That value would act as an approximate lower bound for the timeout, which could then be further padded to reflect physical realities of data transmission and processing.

Reply:
The delay should usually trigger the inbound fluff delay where it's using a poisson distribution, simply mimicking what Bitcoin was doing for the same situation. 95% of the values are in the 3-7.3 second range. A delay of 16 is basically nearly impossibl, so this should be acceptable.

As an explanation of why not exponential here - one of the nodes will have no outbound peers. I don't recall the paper stating how to handle the situation, so I decided to make it immediately fluff, as it would be an edge case. A fluff does randomized poisson delays, similar to what Bitcoin was doing at the time. I don't recall the d++ specifying how to handle fluff precisely either, maybe I need to revisit that paper.

Reply:
@vtnerd Per discussion in https://github.com/monero-project/monero/issues/9755, it might help to make this upper bound a bit more strict. As a 0% failure probability would be impossible for any test (CPUs can fail, etc), I'd aim for something around 1 in 1 million. How much could I reduce the timeout and achieve that bound?

Reply:
Unless I've done the math horribly wrong, it looks like 13s should yield a failure probability of roughly 1 in a billion (I decided a million was too low); I've updated the test timeout accordingly.